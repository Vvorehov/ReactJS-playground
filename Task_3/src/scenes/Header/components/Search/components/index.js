export { SearchButton } from './SearchButton';
export { SearchFilter } from './SearchFilter';
export { SearchField } from './SearchField';
