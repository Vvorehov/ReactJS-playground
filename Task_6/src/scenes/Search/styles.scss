$brandColor: #ED5664;
$secondaryColor: #BDBBBE;

.search {
  padding: 0 6rem 0;
  color: #fff;
}
.search_title {
  font-size: 18px;
  font-weight: 700;
  text-transform: uppercase;
  margin-bottom: 2rem;
}

.search_input {
  display: block;
  width: 100%;
  padding: 1rem 2rem;
  outline: none;
  border: none;
  border-bottom: 2px solid $brandColor;
  border-radius: 2px;
  background-color: rgba(0,0,0,0.8);
  color: #fff;
}

.search_type {
  display: flex;
  align-items: center;
  justify-content: flex-start;
}

.search_type-title {
  padding-right: 2rem;
  text-transform: uppercase;
}

.search_type-btn {
  padding: 0 1rem;
}

.search_submit {
  position: relative;
  display: inline-block;
  width: auto;
  max-width: 100%;
  font-size: inherit;
  padding: .85em 3em;
  line-height: 1;
  border: 0;
  outline: 0;
  border-radius: 2px;
  box-shadow: none;
  text-align: center;
  background-color: $brandColor;
  color: #fff;
  cursor: pointer;
  text-transform: uppercase;
  transition-property: color, background;
  transition-duration: .2s;
  transition-timing-function: ease-out;
  // prevent
  user-select: none;
  appearance: none;
  &:focus {
    outline: 1px solid rgba($brandColor, .5);
  }
  &:hover,
  &:active {
    background-color: darken($brandColor, 10%);
  }
}

.search_footer {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 1rem 0;
}