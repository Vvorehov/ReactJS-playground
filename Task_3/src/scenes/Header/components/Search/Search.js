import React from "react";
import { render } from "react-dom";
import { SearchButton, SearchFilter, SearchField } from "./components";

const Search = () => {
    return <div className="search">
        <SearchField/>
        <div className="search_container">
            <span>Search by</span>
            <SearchFilter />
            <SearchButton />
        </div>
    </div>;
};

export default Search;