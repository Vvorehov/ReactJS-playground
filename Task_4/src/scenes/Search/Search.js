import React, {Component} from 'react';
import { Radio } from '../Radio';
import './styles.scss'

class Search extends Component {
    searchFilms(e) {
        e.preventDefault();
    }

    render() {
        return (
          <form className="search" onSubmit={this.searchFilms.bind(this)}>
              <p className="search_title">Find your movie</p>
              <input type="text" className="search_input" placeholder="search..."/>
              <div className="search_footer">
                  <div className="search_type">
                      <p className="search_type-title">search by</p>
                      <div className="search_type-btn">
                          <Radio name="search-by" label="Title" value="title" isChecked={true}/>
                      </div>
                      <div className="search_type-btn">
                          <Radio name="search-by" label="Genre" value="genre"/>
                      </div>
                  </div>
                  <button className="search_submit" type="submit">Search</button>
              </div>
          </form>
        )
    }
}

export default Search;