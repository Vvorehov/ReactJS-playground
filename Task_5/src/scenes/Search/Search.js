import React, {Component} from 'react';
import { createStore } from 'redux';
import { Radio } from '../Radio';
import actionsTypes from "../../actions/actionsTypes.js";
import './styles.scss'

class Search extends Component {
    searchFilms(e) {
        e.preventDefault();
        let actionType = document.querySelector('.search input[name="search-by"]:checked').getAttribute('data-action-type');

        console.log("actionType: " + actionType);
        const sort = (state = 0, action) => {
            switch (action.type) {
                case actionsTypes.SEARCH_BY_TITLE:
                    return 0;
                case actionsTypes.SEARCH_BY_GENRE:
                    return 1;
                default:
                    return 0;
            }
        };

        const store = createStore(sort);
        store.dispatch({type: actionType});

        store.subscribe(() => {

        })
    }



    render() {
        return (
          <form className="search" onSubmit={this.searchFilms.bind(this)}>
              <p className="search_title">Find your movie</p>
              <input type="text" className="search_input" placeholder="search..."/>
              <div className="search_footer">
                  <div className="search_type">
                      <p className="search_type-title">search by</p>
                      <div className="search_type-btn">
                          <Radio name="search-by" label="Title" onChange="setType()" dataActionType={actionsTypes.SEARCH_BY_TITLE} value="title" isChecked={true}/>
                      </div>
                      <div className="search_type-btn">
                          <Radio name="search-by" label="Genre" dataActionType={actionsTypes.SEARCH_BY_GENRE} value="genre"/>
                      </div>
                  </div>
                  <button className="search_submit" type="submit">Search</button>
              </div>
          </form>
        )
    }
}

export default Search;