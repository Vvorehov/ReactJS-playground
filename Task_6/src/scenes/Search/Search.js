import React, {Component} from 'react';
import { createStore } from 'redux';
import { Radio } from '../Radio';
import actionsTypes from "../../constants/actionsTypes";
import * as FetchData from "../../actions/fetchData";
import { setSearchData } from "../../actions/searchByAction";
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import './styles.scss';

class Search extends Component {
  constructor(props) {
    super(props);
    this.state = {
      searchQuery: props.match.params.searchQuery || props.searchData.searchQuery,
      searchBy: props.match.params.searchBy || props.searchData.searchBy
    };

    this.handleFormSubmit = this.handleFormSubmit.bind(this);
    this.handleOnChange = this.handleOnChange.bind(this);
  }

  componentDidMount() {
    this.props.fetchSearchResults(this.state);
  }

  componentDidUpdate(prevProps) {
    /*if (this.props.location !== prevProps.location) {
      this.props.fetchSearchResults(this.state.searchData);
    }*/
  }

  handleFormSubmit(e) {
    e.preventDefault();
    this.props.fetchSearchResults(this.state);
    this.props.setSearchData(this.state);

    let url = "/search/" + this.state.searchBy.toLowerCase() + "/" + this.state.searchQuery.toLowerCase();
    this.props.history.push(url);
  }

  handleOnChange(e) {
    this.setState({
        [e.target.name]: e.target.value
    })
  }

  render() {
      return (
        <form className="search" onSubmit={this.handleFormSubmit}>
            <p className="search_title">Find your movie</p>
            <input type="text" className="search_input" placeholder="search..." name="searchQuery" onChange={this.handleOnChange} value={this.state.searchQuery}/>
            <div className="search_footer">
                <div className="search_type">
                    <p className="search_type-title">search by</p>
                    <div className="search_type-btn">
                      <Radio name="searchBy" label="Title" value="TITLE" onChange={this.handleOnChange} isChecked={this.state.searchBy.toLowerCase() === "TITLE".toLowerCase()}/>
                    </div>
                    <div className="search_type-btn">
                      <Radio name="searchBy" label="Genre" value="GENRES" onChange={this.handleOnChange} isChecked={this.state.searchBy.toLowerCase() === "GENRES".toLowerCase()}/>
                    </div>
                </div>
                <button className="search_submit" type="submit">Search</button>
            </div>
        </form>
      )
  }
}

function mapStateToProps(state) {
  return {
    searchData: state.searchData
  }
}

function mapDispatchToProps(dispatch) {
  return {
    fetchSearchResults: bindActionCreators(FetchData.fetchSearchResults, dispatch),
    setSearchData: bindActionCreators(setSearchData, dispatch)
  }
}
export default connect(mapStateToProps, mapDispatchToProps)(Search)