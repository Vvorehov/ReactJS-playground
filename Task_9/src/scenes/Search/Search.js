import React, {Component} from 'react';
import { createStore } from 'redux';
import { Radio } from '../Radio';
import actionsTypes from "../../constants/actionsTypes";
import * as FetchData from "../../actions/fetchData";
import { setSearchData } from "../../actions/searchByAction";
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import styled from 'styled-components';
import './styles.scss';

const Button = styled.button.attrs({
  type: 'submit'
})`
    position: relative;
    display: inline-block;
    width: auto;
    max-width: 100%;
    font-size: inherit;
    padding: .85em 3em;
    line-height: 1;
    border: 0;
    outline: 0;
    border-radius: 2px;
    box-shadow: none;
    text-align: center;
    background-color: #ED5664;
    color: #fff;
    cursor: pointer;
    text-transform: uppercase;
    transition-property: color, background;
    transition-duration: .2s;
    transition-timing-function: ease-out;
    user-select: none;
    appearance: none;
    &:hover, &:active {
    background-color: #e8283a;
    }
    &:focus {
      outline: 1px solid rgba(237, 86, 100, 0.5);
    }
  }
`;

class Search extends Component {
  constructor(props) {
    super(props);
    this.state = {
      searchQuery: props.match.params.searchQuery || props.searchData.searchQuery,
      searchBy: props.match.params.searchBy || props.searchData.searchBy
    };

    this.handleFormSubmit = this.handleFormSubmit.bind(this);
    this.handleOnChange = this.handleOnChange.bind(this);
  }

  componentDidMount() {
    this.props.fetchSearchResults(this.state);
  }

  componentDidUpdate(prevProps) {
    /*if (this.props.location !== prevProps.location) {
      this.props.fetchSearchResults(this.state.searchData);
    }*/
  }

  handleFormSubmit(e) {
    e.preventDefault();
    this.props.fetchSearchResults(this.state);
    this.props.setSearchData(this.state);

    let url = "/search/" + this.state.searchBy.toLowerCase() + "/" + this.state.searchQuery.toLowerCase();
    this.props.history.push(url);
  }

  handleOnChange(e) {
    this.setState({
        [e.target.name]: e.target.value
    })
  }

  render() {
      return (
        <form className="search" onSubmit={this.handleFormSubmit}>
            <p className="search_title">Find your movie</p>
            <input type="text" className="search_input" placeholder="search..." name="searchQuery" onChange={this.handleOnChange} value={this.state.searchQuery}/>
            <div className="search_footer">
                <div className="search_type">
                    <p className="search_type-title">search by</p>
                    <div className="search_type-btn">
                      <Radio name="searchBy" label="Title" value="TITLE" onChange={this.handleOnChange} isChecked={this.state.searchBy.toLowerCase() === "TITLE".toLowerCase()}/>
                    </div>
                    <div className="search_type-btn">
                      <Radio name="searchBy" label="Genre" value="GENRES" onChange={this.handleOnChange} isChecked={this.state.searchBy.toLowerCase() === "GENRES".toLowerCase()}/>
                    </div>
                </div>
                {/*<button className="search_submit" type="submit">Search</button>*/}
                <Button>Search</Button>
            </div>
        </form>
      )
  }
}

function mapStateToProps(state) {
  return {
    searchData: state.searchData
  }
}

function mapDispatchToProps(dispatch) {
  return {
    fetchSearchResults: bindActionCreators(FetchData.fetchSearchResults, dispatch),
    setSearchData: bindActionCreators(setSearchData, dispatch)
  }
}
export default connect(mapStateToProps, mapDispatchToProps)(Search)