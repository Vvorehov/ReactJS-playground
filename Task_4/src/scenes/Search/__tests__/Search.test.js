import React from 'react';
import Search from '../Search';
import TestRenderer from 'react-test-renderer';

describe('Search layout complete', () => {
  it('Render correct', () => {

    const component = TestRenderer.create(
      <Search />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
