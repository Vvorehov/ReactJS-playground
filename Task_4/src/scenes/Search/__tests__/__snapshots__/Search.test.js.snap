// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search layout complete Render correct 1`] = `
<form
  className="search"
  onSubmit={[Function]}
>
  <p
    className="search_title"
  >
    Find your movie
  </p>
  <input
    className="search_input"
    placeholder="search..."
    type="text"
  />
  <div
    className="search_footer"
  >
    <div
      className="search_type"
    >
      <p
        className="search_type-title"
      >
        search by
      </p>
      <div
        className="search_type-btn"
      >
        <label
          className="radio"
        >
          <input
            defaultChecked={true}
            name="search-by"
            onChange={[Function]}
            type="radio"
            value="title"
          />
          <span>
            Title
          </span>
        </label>
      </div>
      <div
        className="search_type-btn"
      >
        <label
          className="radio"
        >
          <input
            defaultChecked={undefined}
            name="search-by"
            onChange={[Function]}
            type="radio"
            value="genre"
          />
          <span>
            Genre
          </span>
        </label>
      </div>
    </div>
    <button
      className="search_submit"
      type="submit"
    >
      Search
    </button>
  </div>
</form>
`;
